Update tenplay.py to work with 1080p and no ads

Sipherdrakon opened the pull request:
<!--
    **IMPORTANT**: PRs without the template will be CLOSED
    
    Due to the high volume of pull requests, it may be a while before your PR is reviewed.
    Please try to keep your pull request focused on a single bugfix or new feature.
    Pull requests with a vast scope and/or very large diff will take much longer to review.
    It is recommended for new contributors to stick to smaller pull requests, so you can receive much more immediate feedback as you familiarize yourself with the codebase.

    PLEASE AVOID FORCE-PUSHING after opening a PR, as it makes reviewing more difficult.
-->

### Description of your *pull request* and other information

Not sure if this good enough for yt-dlp but I did test it on

https://10.com.au/australian-survivor/episodes/season-11-australia-v-the-world/episode-1/tpv250814jsbfz
(has 720p original resolution and can get 1080p)

https://10.com.au/the-amazing-race-australia/episodes/season-6/episode-3/tpv250909hsapk
(has 540p original resolution and can get 1080p)

Fixes #14276


<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--
    # PLEASE FOLLOW THE GUIDE BELOW

    - You will be asked some questions, please read them **carefully** and answer honestly
    - Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
    - Use *Preview* tab to see what your *pull request* will actually look like
-->

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check those that apply and remove the others:
- [X] I am the original author of the code in this PR, and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*? Check those that apply and remove the others:
- [X] Fix or improvement to an extractor (Make sure to add/update tests)

</details>

Sipherdrakon committed:
Update tenplay.py to work with 1080p and no ads

Sipherdrakon committed:
Update tenplay.py to deal with CodeQL issues

bashonly reviewed (commented):
did an LLM write this patch?

Sipherdrakon commented:
I had GPT5 adapt my vinetrimmer script (based on ozivine's approach) since every time I tried on my own I would either get a proxy error when trying to use a data url or having to use `--enable-file-urls` in order to grab the 1080p stream from a manipulated m3u8, so figured it might be able to find a way.

Sipherdrakon committed:
Clean up AI Slop (hopefully)

Ran through and edited the code to be cleaner, hopefully it's better now. :-)

Sipherdrakon committed:
Removed Base Headers as they aren't needed.

Sipherdrakon committed:
Found some logic issues that I missed

Ran back through one more time and found some dead code and a few logic issues that needed cleaning.

Sipherdrakon committed:
Added all formats for ad-filtering

This should handle pre-filtering ads for any format found (I couldn't think of a good way to only work it for the specific format called)

Sipherdrakon committed:
Merge branch 'master' into 10play-extractor-fix

dirkf reviewed (commented):
Some suggestions (untested b/c region/account) and questions ...
